modules/seqexec/model/shared/src/main/scala/seqexec/model/ObservationProgress.scala

```diff
@@ -33,6 +33,9 @@ object Progress {
 
   implicit val nsProgressP: Prism[Progress, NSObservationProgress] =
     GenPrism[Progress, NSObservationProgress]
+
+  implicit val progressP: Prism[Progress, Progress] =
```

Comment:
This seems odd? prism form Progress to Progress?

Reply:
I agree, I had my doubts about this.

This comes from the requirement to have a `connect` that provides a generic `Progress` and get the `ObserveStage` to check out if we are if we are `ReadingOut`. It is the first time we need such generic `Progress`, up until now we always requested the `connect` knowing whether we are in a regular observation or a N&S one (which have different `Progress` subclasses).

We obtain the `A <: Progress` `connect` from https://github.com/gemini-hlsw/ocs3/blob/develop/modules/seqexec/web/client/src/main/scala/seqexec/web/client/circuit/SeqexecCircuit.scala#L140, which requires an implicit `Prism[Progress, A]`, so this was the way I found to satisfy  such implicit.

Maybe there's a better way to do it?

Reply:
I see now that an `Iso` seems to be a valid `Prism`. Maybe it would be cleaner to define an `Iso` in this case?

Reply:
I tried: 

```
implicit val progressP: Prism[Progress, Progress] =
    Iso.id[Progress]
```

but it fails to compile with:
```
[error] /mnt/c/dev/gemini/ocs3/modules/seqexec/model/shared/src/main/scala/seqexec/model/ObservationProgress.scala:38:11: type mismatch;
[error]  found   : monocle.Iso[seqexec.model.Progress,seqexec.model.Progress]
[error]     (which expands to)  monocle.PIso[seqexec.model.Progress,seqexec.model.Progress,seqexec.model.Progress,seqexec.model.Progress]
[error]  required: monocle.Prism[seqexec.model.Progress,seqexec.model.Progress]
[error]     (which expands to)  monocle.PPrism[seqexec.model.Progress,seqexec.model.Progress,seqexec.model.Progress,seqexec.model.Progress]
[error]     Iso.id[Progress]
[error]           ^
[info] monocle.Iso[seqexec.model.Progress,seqexec.model.Progress] <: monocle.Prism[seqexec.model.Progress,seqexec.model.Progress]?
[info] false
```

Reply:
Ah, this seems to work:

```
  implicit val progressP: Prism[Progress, Progress] =
    Iso.id[Progress].asPrism
```